src/metabase/models/card.clj

```diff
@@ -101,6 +102,36 @@
   ([_ pk]
    (mi/can-read? (t2/select-one :model/Card :id pk))))
 
+(def ^{:private true
+       :doc     "All acceptable card types.
+                Previously (< 49), we only had 2 card types: question and model, which were differentiated using the
+                boolean `dataset` column. Soon we'll have more card types (e.g: metric) and we will longer be able to use a boolean
+                column to differentiate between all types. So we've added a new `type` column for this purpose.
+
+                Migrating all the code to use `report_card.type` will be quite an effort, we decided that we'll migrate it gradually.
+                In the mean time we'll have both `type` and `dataset` columns."} card-types
+  #{"model" "question" #_:metric})
+
+(def CardTypes
+  "Malli schema for acceptable card types."
+  (into [:enum] card-types))
+
+(mu/defn ^:private is-type? :- :boolean
+  "Check if card is a model?"
+  [target-type :- CardTypes
+   {:keys [type] :as _card} :- [:map [:type CardTypes]]]
+  (= target-type type))
+
+(defn is-question?
```

Comment:
functions that return boolean values conventionally don't have a `is-` prefix. see `metabase.lib.types.isa`
```suggestion
(defn question?
```

Reply:
makes sense, changed.